dwave/plugins/sklearn/transformers.py

```diff
@@ -0,0 +1,292 @@
+    """Select features using a quadratic optimization problem solved on a hybrid solver.
+
+    Args:
+        alpha:
+            Hyperparameter between 0 and 1 that controls the relative weight of
+            the relevance and redundancy terms.  `alpha=1` places all weight on
+            relevance and selects all features, whereas `alpha=0` places all
+            weight on redundancy and selects no features.
+
+        num_features:
+            The number of features to select.
+
+        time_limit:
+            The time limit for the run on the hybrid solver.
+
+    """
+
+    ACCEPTED_METHODS = [
+        "correlation",
+        # "mutual information",  # todo
+        ]
+
+    def __init__(
+        self,
+        *,
+        alpha: float = .5,
+        method: str = "correlation",  # undocumented until there is another supported
+        num_features: int = 10,
+        time_limit: typing.Optional[float] = None,
+    ):
+        if not 0 <= alpha <= 1:
+            raise ValueError(f"alpha must be between 0 and 1, given {alpha}")
+
+        if method not in self.ACCEPTED_METHODS:
+            raise ValueError(
+                f"method must be one of {self.ACCEPTED_METHODS}, given {method}"
+            )
+
+        if num_features <= 0:
+            raise ValueError(f"num_features must be a positive integer, given {num_features}")
+
+        self._alpha = alpha
+        self._method = method
+        self._num_features = num_features
+        self._time_limit = time_limit  # check this lazily
+
+    def __sklearn_is_fitted__(self) -> bool:
+        # used by `check_is_fitted()`
+        try:
+            self._mask
+        except AttributeError:
+            return False
+
+        return True
+
+    def _get_support_mask(self) -> np.ndarray[typing.Any, np.dtype[np.bool_]]:
+        """Get the boolean mask indicating which features are selected
+
+        Returns:
+          boolean array of shape [# input features]. An element is True iff its
+          corresponding feature is selected for retention.
+
+        Raises:
+            RuntimeError: This method will raise an error if it is run before `fit`
+        """
+        check_is_fitted(self)
+
+        try:
+            return self._mask
+        except AttributeError:
+            raise RuntimeError("fit hasn't been run yet")
+
+    def correlation_cqm(
+        self,
+        X: npt.ArrayLike,
+        y: npt.ArrayLike,
+        *,
+        alpha: float,
+        num_features: int,
+        strict: bool = True,
+    ) -> dimod.ConstrainedQuadraticModel:
+        """Build a constrained quadratic model for feature selection.
+
+        This method is based on maximizing influence and independence as
+        measured by correlation [Milne et al.]_.
+
+        Args:
+            X:
+                2D array-like of feature vectors (numerical).
+            y:
+                1D array-like of class labels (numerical).
```

Comment:
```suggestion
                Class labels formatted as a numerical 1D array-like.
```

Very minor but we try to describe what and then how